Sources_v3/StreamChatUI/ChatChannelList/ChatChannelView.swift

```diff
@@ -0,0 +1,140 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+open class ChatChannelView<ExtraData: ExtraDataTypes>: UIView where ExtraData.Channel: NameAndImageProviding {
+    
+    // MARK: - Properties
+    
+    public let uiConfig: UIConfig<ExtraData>
+    
+    public var channel: _ChatChannel<ExtraData>? {
+        didSet { updateAppearance() }
+    }
+    
+    // MARK: - Subviews
+    
+    public private(set) lazy var container: ContainerStackView = {
+        let stack = ContainerStackView()
+        stack.translatesAutoresizingMaskIntoConstraints = false
+        return stack
+    }()
+    
+    public private(set) lazy var channelAvatarView: AvatarView = {
+        let avatar = uiConfig.channelList.avatarView.init()
+        avatar.translatesAutoresizingMaskIntoConstraints = false
+        return avatar
+    }()
+    
+    public private(set) lazy var channelNameLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+    
+    public private(set) lazy var separatorView: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    // MARK: - Subviews Appearance
+
+    @objc public dynamic var channelNameLabelTextColor: UIColor {
+        get { channelNameLabel.textColor }
+        set { channelNameLabel.textColor = newValue }
+    }
+    
+    @objc public dynamic var channelAvatarViewBackgroundColor: UIColor? {
+        get { channelAvatarView.backgroundColor }
+        set { channelAvatarView.backgroundColor = newValue }
+    }
+    
+    @objc public dynamic var separatorColor: UIColor? {
+        get { separatorView.backgroundColor }
+        set { separatorView.backgroundColor = newValue }
+    }
+    
+    // MARK: - Init
+    
+    public required init(
+        channel: _ChatChannel<ExtraData>? = nil,
+        uiConfig: UIConfig<ExtraData> = .default
+    ) {
+        self.channel = channel
+        self.uiConfig = uiConfig
+        
+        super.init(frame: .zero)
+        
+        commonInit()
+    }
+    
+    public required init?(coder: NSCoder) {
+        self.uiConfig = .default
+        self.channel = nil
+        
+        super.init(coder: coder)
+        
+        commonInit()
+    }
+    
+    public func commonInit() {
```

Comment:
can we make this lazy? for example call `setupLayout` when `layoutSubviews` is called?